src/containers/singleEvent/index.tsx

```diff
@@ -101,28 +104,40 @@ interface SingleEventParams {
 const SingleEvent: React.FC<SingleEventProps> = ({
   tokens,
   events,
+  myEvents,
   eventAnnouncements,
 }) => {
   const dispatch = useDispatch();
   const id = Number(useParams<SingleEventParams>().id);
 
   useEffect(() => {
     dispatch(getUpcomingEvents());
+    dispatch(getMyEvents());
     dispatch(getEventAnnouncements(id));
   }, [dispatch, id]);
 
   const privilegeLevel: PrivilegeLevel = useSelector((state: C4CState) => {
     return getPrivilegeLevel(state.authenticationState.tokens);
   });
 
+  const hasRegistered: boolean = useSelector((state: C4CState) => {
+    return (
+      asyncRequestIsComplete(state.myEventsState.myEvents) &&
+      state.myEventsState.myEvents.result.some((e) => e.id === id)
+    );
```

Comment:
🌱 I think this is a good case for defining a true selector, ie a util function that does this for us, so that we can test it and our line here becomes `const hasRegistered: boolean = useSelector(state => getHasRegistered(id));`